build.gradle

```diff
@@ -0,0 +1,178 @@
+                  PLUGIN_ID: pluginId,
+                  PLUGIN_NAME: pluginName
+               ])
+      }
+      println "Complete, folder created at $pluginsDir/$pluginName"
+    } catch(FileAlreadyExistsException e) {
+      throw new GradleScriptException("Plugin folder for '$pluginName' already exists", e)
+    }
+  }
+}
+
+task verifyPlugin {
+  description "Validates the contents of a plugin directory"
+  
+  doFirst {
+    if(!path) {
+      throw new IllegalArgumentException("Missing required parameter 'pluginPath'")
+    }
+    dir = Paths.get(path).toAbsolutePath()
+    if(!Files.exists(dir) || !Files.isDirectory(dir)) {
+      throw new IllegalArgumentException("The path does not exist or is not a folder: $dir")
+    }
+    def manifest = dir.resolve("plugin.xml")
+    if(!Files.exists(manifest)) {
+      throw new GradleScriptException("Missing required file ${manifest}", null)
+    }
+    // TODO: Add XML validation
+    Files.newInputStream(manifest).withCloseable { inputStream ->
+      def xml = new XmlParser().parse(inputStream)
+      pluginId = pluginId ?: xml.id.text()
+      pluginName = pluginName ?: xml.name.text()
+      pluginVersion = pluginVersion ?: xml.version.text()
+      crafterEdition = crafterEdition ?: xml['crafter-edition'].text()
+      buildNumber = buildNumber ?: UUID.randomUUID() as String
+    }
+    if(!pluginId || !pluginName || !pluginVersion || !crafterEdition || !buildNumber) {
+      throw new GradleScriptException("Missing required information from manifest file", null)
+    }
+  }
+  
+  doLast {
+    expectedContent().each {
+      def file = Paths.get(it.value in String? it.value : it.value.join(File.separator))
+      if(!Files.exists(dir.resolve(file))) {
+        println "Missing file or folder: '$file'"
+      }
+    }
+    copy {
+      from dir
+      into tempDir
+      filter(ReplaceTokens,
+             tokens: [ 
+                PLUGIN_ID: pluginId,
+                PLUGIN_NAME: pluginName, 
+                PLUGIN_VERSION: pluginVersion,
+                CRAFTER_EDITION: crafterEdition,
+                BUILD_DATE: Instant.now() as String,
+                BUILD_NUMBER: buildNumber
+             ])
+    }
+    // Update configuration for packaging after reading the manifest
+    def zip = tasks['packagePlugin']
+    zip.baseName = pluginName
+    zip.version = pluginVersion
+  }
+}
+
+task packagePlugin(type: Zip, dependsOn: verifyPlugin) {
+  description "Package a Crafter Studio plugin"
+  
+  from tempDir
+  destinationDir pluginsDir
+  extension 'car'
+  
+  doLast {
+    delete tempDir
+    println "Complete, plugin written to file '$pluginsDir${File.separator}$pluginName-${pluginVersion}.car'"
+  }
+}
+
+task publishPlugin {
+  description "Publishes a plugin to the Crafter Marketplace"
+  
+  doLast {
+    if(autoPublish) {
+      println project.getProperty("publish.message.auto")
+      // ant.input(message: 'Pelase type your username', addproperty: 'username')
```

Comment:
`Please` (typo)

Also, shouldn't this indicate that you're asking for github creds?